packages/build-utils/src/convert-runtime-to-plugin.ts

```diff
@@ -150,19 +141,38 @@ export function convertRuntimeToPlugin(
         }
       }
 
-      const handlerFilePath = Object.keys(lambdaFiles).find(item => {
-        return parse(item).name === handlerFileName;
+      let handlerFilePath = Object.keys(lambdaFiles).find(item => {
+        return parse(item).name === output.handler;
       });
 
+      const { handler } = output;
+      const handlerMethod = handler.split('.').reverse()[0];
+      const handlerFileName = handler.replace(`.${handlerMethod}`, '');
+
+      // For compiled languages, the launcher file for the Lambda generated
+      // by the Legacy Runtime matches the `handler` defined for it, but for
+      // interpreted languages, the `handler` consists of the launcher file name
+      // without an extension, plus the name of the method inside of that file
+      // that should be invoked, so we have to construct the file path explicitly.
+      if (!handlerFilePath) {
+        handlerFilePath = Object.keys(lambdaFiles).find(item => {
+          return parse(item).name === handlerFileName;
+        });
```

Comment:
Instead of using `Object.keys` to iterate over all the files again, couldn't you do `lambdaFiles[completeHandlerFileName]`, where `completeHandlerFileName` is also including the path to the handler file?

Reply:
because if we just compare the name, then it's probably pretty easy to also get false-positives, e.g. if there are two files that are called `test`, it could return for either, no?

Reply:
Hmm, but I guess `completeHandlerFileName` is what you're trying to figure out? But shouldn't there be a specific path that we can check since we know the source file name?

Reply:
`handlerFileName` doesn't have an extension, and we don't know which one it has before we find the actual file that's associated with it, so we couldn't construct `completeHandlerFileName`.

We could make the assumption that the `ext` that Runtime authors pass to the utility always matches the handler's extension, but I'm not sure if that's safe. I don't think so, because it would mean assuming that the outputs have the same extensions as the inputs.

Reply:
**TLDR:** We don't know the file name of the handler on the file system, unless we make the assumption mentioned above.

Reply:
I assume it'd be something like this:

```
lambdaFiles[join(entryDir, entryBase, handlerFileName)];
```

Or without the `entryDir`.

Reply:
`handlerFileName` does not contain the extension. The `handler` that Runtimes return for their `Lambda` types does not contain the extension, so we don't know it before iterating through `lambdaFiles`.

Reply:
> handlerFileName doesn't have an extension, and we don't know which one it has before we find the actual file that's associated with it, so we couldn't construct completeHandlerFileName.

I see

Reply:
But just comparing the file name won't be enough.

We'll have to include the path, and the path should be known, right?

Reply:
If you think it's safe to rely on `ext` we can do it — I don't think it's safe, but I also don't know of any examples where it would break.

Reply:
Though, if we don't know the extension, it could still be a false-positive if there are two files with a different extension in the same directory.

Reply:
Agreed that we probably want to rely on `ext`, otherwise, we'll have the problem of two files with the same name (without an extension) causing a problem. I'll fix that.

Reply:
If the filename is anything other than `vc___` or something similar, then it's definitely not safe to rely on the filename either. Probably less than the extension.

Reply:
I also just noticed that this doesn't support `.exe` handlers because those are missing the extension on the left side, but they have their extension on the right side.

Reply:
Fixed!